gcp_variant_transforms/transforms/merge_headers_test.py

```diff
@@ -300,6 +304,104 @@ def test_combine_two_headers_with_bad_conflict(self):
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample3\n']
+
+    vcf_reader_1 = vcf.Reader(fsock=iter(lines_1))
+    vcf_reader_2 = vcf.Reader(fsock=iter(lines_2))
+    headers_1 = self._get_header_from_reader(vcf_reader_1)
+    headers_2 = self._get_header_from_reader(vcf_reader_2)
+    combiner_fn = self._get_combiner_fn(save_conflicts=True)
+
+    merged_headers = combiner_fn.create_accumulator()
+    merged_headers = combiner_fn.add_input(merged_headers, headers_1)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_2)
+    merged_headers = combiner_fn.extract_output(merged_headers)
+
+    self.assertItemsEqual(merged_headers.conflicts['NS'],
+                          set(['num=1 type=Float']))
+    self.assertItemsEqual(merged_headers.conflicts['DP'],
+                          set(['num=1 type=Integer']))
+
+  def test_merge_and_save_conflicts_three_conflicting_headers(self):
+    lines_1 = [
+        '##fileformat=VCFv4.2\n',
+        '##INFO=<ID=NS,Number=2,Type=Integer,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample1 Sample2\n']
+    lines_2 = [
+        '##fileformat=VCFv4.2\n',
+        '##INFO=<ID=NS,Number=2,Type=Float,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample3\n']
+    lines_3 = [
+        '##fileformat=VCFv4.2\n',
+        '##INFO=<ID=NS,Number=.,Type=Float,Description="Number samples">\n',
+        '#CHROM  POS ID  REF ALT QUAL  FILTER  INFO  FORMAT  Sample3\n']
+
+    vcf_reader_1 = vcf.Reader(fsock=iter(lines_1))
+    vcf_reader_2 = vcf.Reader(fsock=iter(lines_2))
+    vcf_reader_3 = vcf.Reader(fsock=iter(lines_3))
+    headers_1 = self._get_header_from_reader(vcf_reader_1)
+    headers_2 = self._get_header_from_reader(vcf_reader_2)
+    headers_3 = self._get_header_from_reader(vcf_reader_3)
+    combiner_fn = self._get_combiner_fn(save_conflicts=True)
+
+    merged_headers = combiner_fn.create_accumulator()
+    merged_headers = combiner_fn.add_input(merged_headers, headers_1)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_2)
+    merged_headers = combiner_fn.add_input(merged_headers, headers_3)
+    merged_headers = combiner_fn.extract_output(merged_headers)
+
+    self.assertItemsEqual(merged_headers.infos.keys(), ['NS'])
+    self.assertItemsEqual(merged_headers.infos['NS'],
+                          OrderedDict([('id', 'NS'),
+                                       ('num', None),
+                                       ('type', 'Float'),
+                                       ('desc', 'Number samples'),
+                                       ('source', None),
+                                       ('version', None)]))
+    self.assertItemsEqual(merged_headers.conflicts['NS'],
+                          set(['num=2 type=Float', 'num=2 type=Integer',
+                               'num=None type=Float']))
+
```

Comment:
Could you also test the case with missing definition (a field is define in only one file, and missing in the other)?